WireDomain/Tests/WireDomainTests/Event Processing/TeamEventProcessor/TeamMemberLeaveEventProcessorTests.swift

```diff
@@ -0,0 +1,331 @@
+//
+// Wire
+// Copyright (C) 2024 Wire Swiss GmbH
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import Foundation
+import WireAPI
+import WireAPISupport
+import WireDataModel
+import WireDataModelSupport
+import WireDomainSupport
+import XCTest
+
+@testable import WireDomain
+
+final class TeamMemberLeaveEventProcessorTests: XCTestCase {
+
+    var sut: TeamMemberLeaveEventProcessor!
+
+    var coreDataStack: CoreDataStack!
+    let coreDataStackHelper = CoreDataStackHelper()
+    let modelHelper = ModelHelper()
+    var repository: TeamRepositoryProtocol!
+
+    var context: NSManagedObjectContext {
+        coreDataStack.syncContext
+    }
+
+    override func setUp() async throws {
+        coreDataStack = try await coreDataStackHelper.createStack()
+
+        repository = TeamRepository(
+            selfTeamID: UUID(),
+            userRepository: UserRepository(context: context, usersAPI: MockUsersAPI()),
+            teamsAPI: MockTeamsAPI(),
+            context: context
+        )
+
+        sut = TeamMemberLeaveEventProcessor(repository: repository)
+        try await super.setUp()
+    }
+
+    override func tearDown() async throws {
+        coreDataStack = nil
+        sut = nil
+        repository = nil
+        try coreDataStackHelper.cleanupDirectory()
+        try await super.tearDown()
+    }
+
+    // MARK: - Tests
+
+    func testProcessEvent_It_Deletes_A_Member_When_Receiving_A_Team_Member_Leave_Update_Event_For_Another_User() async throws {
+        // Given
+
+        await context.perform { [context] in
+            let user = ZMUser.insertNewObject(in: context)
+            user.remoteIdentifier = Scaffolding.userID
+
+            let team = Team.insertNewObject(in: context)
+            team.remoteIdentifier = Scaffolding.teamID
```

Comment:
Maybe it's worth expanding the `ModelHelper` to include these. You could even define a helper method that creates the user(s), team, and membership in a single method.